ipn/ipnlocal/local.go

```diff
@@ -6453,8 +6453,16 @@ func suggestExitNode(report *netcheck.Report, netMap *netmap.NetworkMap, r *rand
 	if report.PreferredDERP == 0 {
 		return res, ErrNoPreferredDERP
 	}
+	var allowedCandidates set.Set[string]
+	if allowed, ok := syspolicy.GetStringArray(syspolicy.AllowedSuggestedExitNodes, []string{}); ok == nil {
+		allowedCandidates = set.SetOf(allowed)
+	}
 	candidates := make([]tailcfg.NodeView, 0, len(netMap.Peers))
+	isAuto := netMap.SelfNode.CapMap().Has(tailcfg.NodeAttrAutoExitNode)
 	for _, peer := range netMap.Peers {
+		if isAuto && allowedCandidates != nil && !allowedCandidates.Contains(string(peer.StableID())) {
+			continue
+		}
```

Comment:
Actually, I think we should drop the `isAuto` check -- I think we should honor the policy always, and if people are using MDM features that their plan doesn't allow for, that's a sales conversation rather than requiring hard enforcement. This way conflates the auto exit node feature with the ability to restrict the suggestion list.